src/interfaces/noise_model.cpp

```diff
@@ -0,0 +1,19 @@
+#include "gl_depth_sim/interfaces/noise_model.h"
+
+std::vector<float> gl_depth_sim::noise(std::vector<float> distance)
+{
+  float segma;
+  std::random_device r;
+  std::seed_seq seed2{r(), r(), r(), r(), r(), r(), r(), r()};
+  std::mt19937 e2(seed2);
+
+  for (int i = 0 ; i < distance.size() ;++i){
+  // Seed with a real random value, if available
+    segma= 0.0012+0.0019* pow(distance[i]-0.4,2);
+    float r = distance[i];
+    // Generate a normal distribution around that mean
+    std::normal_distribution<> normal_dist(r, segma);
```

Comment:
Likewise this line creates a normal distribution that is of type double. The `<>` brackets indicate that it is of the default template type. If you look at the docs [here](https://en.cppreference.com/w/cpp/numeric/random/normal_distribution) you'll see that means double. This could actually make difference because the random number generator will be asking for twice the random bits that it absolutely needs. Consider specifying the distribution as having template type float: `std::normal_distribution<float>`.